### PR TITLE
Port gen_send_iseq to the new backend IR

### PR DESCRIPTION
**Pull request by k0kubun**

### CODE REVIEW

File: yjit/src/codegen.rs [hunk 1]
```diff
@@ -4367,13 +4362,13 @@ fn gen_send_iseq(
...
+    asm.comment("stack overflow check");
     let stack_max: i32 = unsafe { get_iseq_body_stack_max(iseq) }.try_into().unwrap();
     let locals_offs =
         (SIZEOF_VALUE as i32) * (num_locals + stack_max) + 2 * (RUBY_SIZEOF_CONTROL_FRAME as i32);
-    lea(cb, REG0, ctx.sp_opnd(locals_offs as isize));
-    cmp(cb, REG_CFP, REG0);
-    jle_ptr(cb, counted_exit!(ocb, side_exit, send_se_cf_overflow));
+    let stack_limit = asm.lea(ctx.sp_opnd(locals_offs as isize));
+    asm.cmp(CFP, stack_limit);
+    asm.jbe(counted_exit!(ocb, side_exit, send_se_cf_overflow).into());
```
Comment (noahgibbs) [inline yjit/src/codegen.rs:4371]: Oh nice! Didn't realise we had a .into() for that, but that's a lot prettier.

File: yjit/src/codegen.rs [hunk 2]
```diff
@@ -4509,80 +4503,70 @@ fn gen_send_iseq(
...
     // sp[-2] = block_handler;
     match block {
         Some(_) => {
             // reg1 = VM_BH_FROM_ISEQ_BLOCK(VM_CFP_TO_CAPTURED_BLOCK(reg_cfp));
-            lea(cb, REG1, mem_opnd(64, REG_CFP, RUBY_OFFSET_CFP_SELF));
-            or(cb, REG1, imm_opnd(1));
-            mov(cb, mem_opnd(64, REG0, 8 * -2), REG1);
+            let block_handler = asm.lea(Opnd::mem(64, CFP, RUBY_OFFSET_CFP_SELF));
+            let block_handler = asm.or(block_handler, 1.into());
+            asm.mov(Opnd::mem(64, callee_sp, 8 * -2), block_handler);
```
Comment (noahgibbs) [inline yjit/src/codegen.rs]: Minor thing that was true before, but... Does the 8 * -2 want to be SIZEOF_VALUE * -2 here? It's SP-based, so I assume it's based on VALUEs.

**Review by noahgibbs — APPROVED** [review]

**Comment by noahgibbs** [comment]
In general, looks really good. The only problems I'm seeing are from before your change :-)

(And they're minor.)

**Comment by kddnewton** [comment]
Code looks good. Can you change the `asm.mov` invocations to use `asm.store` when you're using a memory operand as the destination though? I'm on a mission to remove that encoding since I worry it'll be confusing in the future.

File: yjit/src/codegen.rs (hunk 2 above)
Comment (maximecb) [reply yjit/src/codegen.rs]: I think it probably is SIZEOF_VALUE, but that code was written before we had a constant for that hehe

Comment (maximecb) [reply yjit/src/codegen.rs]: There's a couple of occurrences of 8 that all seem to be SIZEOF_VALUE and should probably be changed for readability

Comment (maximecb) [reply yjit/src/codegen.rs]: I made the SIZEOF_VALUE edits so we can merge this.

**Comment by k0kubun** [comment]
> Code looks good. Can you change the asm.mov invocations to use asm.store when you're using a memory operand as the destination though? I'm on a mission to remove that encoding since I worry it'll be confusing in the future.

sure, filed https://github.com/Shopify/ruby/pull/385

**Pull request merged**